sql: TestDropTableInterleavedDeleteData failing

make stress PKG=./pkg/sql TESTS=TestDropTableInterleavedDeleteData

fails

```
go test  -exec 'stress ' -tags ' make x86_64_apple_darwin16.7.0' -ldflags '-X github.com/cockroachdb/cockroach/pkg/build.typ=development -extldflags "" -X "github.com/cockroachdb/cockroach/pkg/build.tag=v2.2.0-alpha.20181119-364-g531c19d104" -X "github.com/cockroachdb/cockroach/pkg/build.rev=531c19d1048156cef3a9536025c959457d55a1d5" -X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=x86_64-apple-darwin16.7.0"  ' -run "TestDropTableInterleavedDeleteData" -timeout 0 ./pkg/sql  -v -args -test.timeout 8m
19 runs so far, 0 failures, over 5s
41 runs so far, 0 failures, over 10s
63 runs so far, 0 failures, over 15s

I181128 15:00:23.515392 1 rand.go:75  Random seed: -2212788393210144075
=== RUN   TestDropTableInterleavedDeleteData
W181128 15:00:23.546153 12 server/status/runtime.go:295  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I181128 15:00:23.566598 12 server/server.go:848  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I181128 15:00:23.567080 12 base/addr_validation.go:279  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I181128 15:00:23.567115 12 base/addr_validation.go:319  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I181128 15:00:23.571461 12 server/config.go:483  [n?] 1 storage engine initialized
I181128 15:00:23.571485 12 server/config.go:486  [n?] RocksDB cache size: 128 MiB
I181128 15:00:23.571494 12 server/config.go:486  [n?] store 0: in-memory, size 0 B
I181128 15:00:23.575325 12 util/stop/stopper.go:537  [n?] quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I181128 15:00:23.575468 12 server/node.go:370  [n?] **** cluster 5285211d-3b64-437e-9627-d977817be53c has been created
I181128 15:00:23.575492 12 server/server.go:1394  [n?] **** add additional nodes by specifying --join=127.0.0.1:58369
I181128 15:00:23.575919 12 gossip/gossip.go:393  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:58369" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:1 patch:0 unstable:3 > build_tag:"v2.2.0-alpha.20181119-364-g531c19d104" started_at:1543417223575742000 
I181128 15:00:23.576853 12 server/node.go:474  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=7.1 KiB), ranges=1, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=7227.00 p25=7227.00 p50=7227.00 p75=7227.00 p90=7227.00 pMax=7227.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I181128 15:00:23.576930 12 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I181128 15:00:23.577071 12 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I181128 15:00:23.577984 12 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "5285211d-3b64-437e-9627-d977817be53c"
I181128 15:00:23.578021 12 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I181128 15:00:23.578141 12 server/server.go:1820  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I181128 15:00:23.578335 12 server/server.go:1526  [n1] starting https server at 127.0.0.1:58370 (use: 127.0.0.1:58370)
I181128 15:00:23.578360 12 server/server.go:1528  [n1] starting grpc/postgres server at 127.0.0.1:58369
I181128 15:00:23.578377 12 server/server.go:1529  [n1] advertising CockroachDB node at 127.0.0.1:58369
W181128 15:00:23.578407 12 jobs/registry.go:317  [n1] unable to get node liveness: node not in the liveness table
I181128 15:00:23.587340 49 storage/replica_command.go:325  [n1,split,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I181128 15:00:23.591472 205 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I181128 15:00:23.602202 145 storage/replica_command.go:325  [n1,split,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I181128 15:00:23.607077 229 storage/replica_command.go:325  [n1,split,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I181128 15:00:23.612684 224 storage/replica_command.go:325  [n1,split,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I181128 15:00:23.618978 247 storage/replica_command.go:325  [n1,split,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I181128 15:00:23.635604 274 storage/replica_command.go:325  [n1,split,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I181128 15:00:23.635982 221 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:2.1-3 User:root}
I181128 15:00:23.644977 293 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I181128 15:00:23.651479 298 storage/replica_command.go:325  [n1,split,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I181128 15:00:23.660731 283 storage/replica_command.go:325  [n1,split,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I181128 15:00:23.666967 325 storage/replica_command.go:325  [n1,split,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I181128 15:00:23.674187 278 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:2515d7a7-240e-4643-a6c3-09ada02fbf9d User:root}
I181128 15:00:23.674875 343 storage/replica_command.go:325  [n1,split,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I181128 15:00:23.679367 349 storage/replica_command.go:325  [n1,split,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I181128 15:00:23.683822 358 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I181128 15:00:23.684604 367 storage/replica_command.go:325  [n1,split,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I181128 15:00:23.692693 312 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I181128 15:00:23.693927 386 storage/replica_command.go:325  [n1,split,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I181128 15:00:23.694521 12 server/server.go:1582  [n1] done ensuring all necessary migrations have run
I181128 15:00:23.694548 12 server/server.go:1585  [n1] serving sql connections
I181128 15:00:23.697859 405 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:58369} Attrs: Locality: ServerVersion:2.1-3 BuildTag:v2.2.0-alpha.20181119-364-g531c19d104 StartedAt:1543417223575742000 LocalityAddress:[]} ClusterID:5285211d-3b64-437e-9627-d977817be53c StartedAt:1543417223575742000 LastUp:1543417223575742000}
I181128 15:00:23.699987 402 server/server_update.go:68  [n1] no need to upgrade, cluster already at the newest version
I181128 15:00:23.700689 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.700808 395 storage/replica_command.go:325  [n1,split,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I181128 15:00:23.704061 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704243 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704386 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704518 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704651 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704772 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.704902 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705047 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705185 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705311 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705437 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705610 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705732 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.705914 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.706074 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.706262 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.706473 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.706741 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.707169 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.707878 12 server/testserver.go:421  had 14 ranges at startup, expected 20
I181128 15:00:23.712145 12 server/testserver.go:421  had 15 ranges at startup, expected 20
I181128 15:00:23.712321 400 storage/replica_command.go:325  [n1,split,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I181128 15:00:23.717013 12 server/testserver.go:421  had 16 ranges at startup, expected 20
I181128 15:00:23.718121 445 storage/replica_command.go:325  [n1,split,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I181128 15:00:23.722471 414 storage/replica_command.go:325  [n1,split,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I181128 15:00:23.723824 12 server/testserver.go:421  had 17 ranges at startup, expected 20
I181128 15:00:23.726919 425 storage/replica_command.go:325  [n1,split,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I181128 15:00:23.732455 498 storage/replica_command.go:325  [n1,split,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I181128 15:00:23.733808 12 server/testserver.go:421  had 19 ranges at startup, expected 20
I181128 15:00:23.768273 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "create_database", target: 52, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I181128 15:00:23.771709 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "create_table", target: 53, info: {TableName:t.public.kv Statement:CREATE TABLE kv (k INT PRIMARY KEY, v INT) User:root}
I181128 15:00:23.772918 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=53 (kv) version=2 mtime=2018-11-28 15:00:23.771766 +0000 UTC
I181128 15:00:23.774723 533 storage/replica_command.go:325  [n1,split,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/53 [r21]
I181128 15:00:23.779094 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "create_table", target: 54, info: {TableName:t.public.intlv Statement:CREATE TABLE intlv (k INT, m INT, n INT, PRIMARY KEY (k, m)) INTERLEAVE IN PARENT kv (k) User:root}
I181128 15:00:23.786555 512 storage/replica_command.go:325  [n1,split,s1,r21/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r22]
I181128 15:00:23.787440 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=53 (kv) version=3 mtime=2018-11-28 15:00:23.784549 +0000 UTC
I181128 15:00:23.787705 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=54 (intlv) version=2 mtime=2018-11-28 15:00:23.784549 +0000 UTC
W181128 15:00:23.800894 553 storage/intent_resolver.go:675  [n1,s1] failed to push during intent resolution: failed to push "split" id=2746079b key=/Local/Range/Table/53/RangeDescriptor rw=true pri=0.00324706 iso=SERIALIZABLE stat=PENDING epo=0 ts=1543417223.786569000,0 orig=1543417223.786569000,0 max=1543417223.786569000,0 wto=false rop=false seq=1
I181128 15:00:23.804444 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=53 (kv) version=3 mtime=2018-11-28 15:00:23.785163 +0000 UTC
I181128 15:00:23.804712 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=54 (intlv) version=2 mtime=2018-11-28 15:00:23.785163 +0000 UTC
I181128 15:00:23.823373 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "create_index", target: 54, info: {TableName:t.public.intlv IndexName:intlv_idx Statement:CREATE INDEX intlv_idx ON intlv (k, n) INTERLEAVE IN PARENT kv (k) User:root MutationID:1}
I181128 15:00:23.824931 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "create_index", target: 54, info: {TableName:t.public.intlv IndexName:intlv_idx Statement:CREATE INDEX intlv_idx ON intlv (k, n) INTERLEAVE IN PARENT kv (k) User:root MutationID:1}
I181128 15:00:23.825747 562 sql/lease.go:876  new lease: 53("kv") ver=3:1543417472.833701538,0, refcount=0
I181128 15:00:23.826929 463 sql/lease.go:315  publish (1 leases): desc=[{kv 53 2}]
I181128 15:00:23.859246 140 sql/lease.go:345  [n1,scExec] publish: descID=54 (intlv) version=3 mtime=2018-11-28 15:00:23.859138 +0000 UTC
I181128 15:00:23.861492 140 sql/backfill.go:132  [n1,scExec] Running backfill for "intlv", v=3, m=1
I181128 15:00:23.919009 140 sql/lease.go:345  [n1,scExec] publish: descID=54 (intlv) version=4 mtime=2018-11-28 15:00:23.918875 +0000 UTC
I181128 15:00:23.927381 140 sql/event_log.go:126  [n1,scExec] Event: "finish_schema_change", target: 54, info: {MutationID:1}
I181128 15:00:23.929143 528 sql/lease.go:876  new lease: 54("intlv") ver=4:1543417555.810796509,0, refcount=0
I181128 15:00:23.931763 140 sql/lease.go:315  publish (1 leases): desc=[{intlv 54 3}]
I181128 15:00:24.162344 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=53 (kv) version=4 mtime=2018-11-28 15:00:24.148053 +0000 UTC
I181128 15:00:24.162982 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root] publish: descID=54 (intlv) version=5 mtime=2018-11-28 15:00:24.148053 +0000 UTC
I181128 15:00:24.167691 463 sql/event_log.go:126  [n1,client=127.0.0.1:58371,user=root] Event: "drop_table", target: 54, info: {TableName:t.public.intlv Statement:DROP TABLE t.public.intlv User:root CascadeDroppedViews:[]}
I181128 15:00:24.169859 463 sql/lease.go:315  publish (1 leases): desc=[{kv 53 3}]
I181128 15:00:24.172965 637 sql/lease.go:876  new lease: 53("kv") ver=4:1543417596.027932125,0, refcount=0
I181128 15:00:24.194687 140 sql/lease.go:345  [n1,scExec] publish: descID=54 (intlv) version=6 mtime=2018-11-28 15:00:24.194582 +0000 UTC
I181128 15:00:24.194688 463 sql/lease.go:345  [n1,client=127.0.0.1:58371,user=root,scExec] publish: descID=54 (intlv) version=6 mtime=2018-11-28 15:00:24.194583 +0000 UTC
I181128 15:00:24.204637 140 sql/lease.go:345  [n1,scExec] publish: descID=54 (intlv) version=6 mtime=2018-11-28 15:00:24.194583 +0000 UTC
W181128 15:00:24.321515 140 sql/schema_changer.go:796  [n1,scExec] ID 54 is not a table
W181128 15:00:24.323214 463 sql/exec_util.go:941  [n1,client=127.0.0.1:58371,user=root] error executing schema change: ID 54 is not a table
I181128 15:00:24.323470 12 util/stop/stopper.go:537  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      node.Node: batch
1      [async] storage.merge: processing replica
1      [async] closedts-rangefeed-subscriber
I181128 15:00:24.323718 12 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] storage.merge: processing replica
1      [async] closedts-rangefeed-subscriber
I181128 15:00:24.323879 606 rpc/nodedialer/nodedialer.go:91  [n1,merge,s1,r21/1:/Table/5{3-4}] unable to connect to n1: initial connection heartbeat failed: rpc error: code = Canceled desc = context canceled
I181128 15:00:24.323947 12 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
--- FAIL: TestDropTableInterleavedDeleteData (0.81s)
    drop_test.go:1015: pq: ID 54 is not a table
FAIL


ERROR: exit status 1

66 runs completed, 1 failures, over 16s
context canceled
FAIL
```

--- a/pkg/sql/drop_test.go
+++ b/pkg/sql/drop_test.go
@@ -19,6 +19,7 @@ import (
 	gosql "database/sql"
 	"fmt"
 	"math/rand"
+	"sync/atomic"
 	"testing"
 
 	"github.com/lib/pq"
@@ -994,10 +995,17 @@ func TestDropTableWhileUpgradingFormat(t *testing.T) {
 func TestDropTableInterleavedDeleteData(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	params, _ := tests.CreateTestServerParams()
+	var enableAsync uint32
 	params.Knobs = base.TestingKnobs{
 		SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
 			// Turn on quick garbage collection.
 			AsyncExecQuickly: true,
+			AsyncExecNotification: func() error {
+				if atomic.LoadUint32(&enableAsync) == 0 {
+					return errors.New("async schema changes are disabled")
+				}
+				return nil
+			},
 		},
 	}
 	s, sqlDB, kvDB := serverutils.StartServer(t, params)
@@ -1022,6 +1030,8 @@ func TestDropTableInterleavedDeleteData(t *testing.T) {
 	) {
 		t.Fatalf("different error than expected: %v", err)
 	}
+
+	atomic.StoreUint32(&enableAsync, 1)
 
 	testutils.SucceedsSoon(t, func() error {
 		return descExists(sqlDB, false, tableDescInterleaved.ID)

